experiments/compositions/composition/expanders/jinja2/main.go

```diff
@@ -0,0 +1,169 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"os/exec"
+
+	pb "google.com/composition/proto"
+	"google.golang.org/grpc"
+	"tailscale.com/atomicfile"
+)
+
+var (
+	port = flag.Int("port", 50051, "The server port")
+)
+
+// server is used to implement exander.Evaluator
+type server struct {
+	pb.UnimplementedExpanderServer
+}
+
+type expander struct {
+	path string
+	req  *pb.EvaluateRequest
+}
+
+// Verify the expander config/template
+func (s *server) Validate(context.Context, *pb.ValidateRequest) (*pb.ValidateResult, error) {
+	return &pb.ValidateResult{Status: pb.Status_SUCCESS}, nil
+}
+
+// Evaluate the expander config in context of inputs and return manifests
+func (s *server) Evaluate(ctx context.Context, req *pb.EvaluateRequest) (*pb.EvaluateResult, error) {
+	result := &pb.EvaluateResult{
+		Status:    pb.Status_SUCCESS,
+		Errors:    []string{},
+		Type:      pb.ResultType_MANIFESTS,
+		Manifests: []byte{},
+		Values:    []byte{},
+	}
+
+	log.Printf("EvaluateRequest:\n  config: %s\n  context: %s\n  facade: %s\n  values: %s\n", req.Config, req.Context, req.Facade, req.Value)
+	dir, err := os.MkdirTemp("", "jinja2")
+	if err != nil {
+		newerr := fmt.Errorf("Unexpected tmp file creation failure. %v", err)
+		log.Print(newerr.Error())
+		return nil, newerr
+	}
+	defer os.RemoveAll(dir) // clean up
+
+	e := expander{path: dir, req: req}
+	err = e.WriteInputsToFileSystem()
+	if err != nil {
+		newerr := fmt.Errorf("Error processing inputs: %v", err)
+		log.Print(newerr.Error())
+		return nil, newerr
+	}
+
+	return e.Evaluate(result)
+}
+
+func (e *expander) WriteInputsToFileSystem() error {
+	context := map[string]interface{}{}
+	if string(e.req.Context) != "" {
+		err := json.Unmarshal(e.req.Context, &context)
+		if err != nil {
+			return fmt.Errorf("Error unmarshalling req.Context: %v", err)
+		}
+	}
+
+	facade := map[string]interface{}{}
+	err := json.Unmarshal(e.req.Facade, &facade)
+	if err != nil {
+		return fmt.Errorf("Error unmarshalling req.Facade: %v", err)
+	}
+
+	getterValues := map[string]interface{}{}
+	if string(e.req.Value) != "" {
+		err := json.Unmarshal(e.req.Value, &getterValues)
+		if err != nil {
+			return fmt.Errorf("Error unmarshalling req.Values: %v", err)
+		}
+	}
+
+	valuesObj := map[string]interface{}{
+		"context":      context,
+		e.req.Resource: facade,
+		"values":       getterValues,
+	}
+
+	// marshall values
+	values, err := json.Marshal(valuesObj)
+	if err != nil {
+		return fmt.Errorf("failed to marshal values: %v", err)
+	}
+
+	// Write values to file
+	err = atomicfile.WriteFile(e.path+"/values", values, 0644)
```

Comment:
Nit: should get into the habit of using `filepath.Join` here

Reply:
replaced all concats with filepath.Join()